actionview/lib/action_view/routing_url_for.rb

```diff
@@ -147,5 +135,9 @@ def optimize_routes_generation? #:nodoc:
       def _generate_paths_by_default
         true
       end
+
+      def get_only_path(host)
+        host.nil? ? _generate_paths_by_default : false
```

Comment:
@abhishekjain16 this is something you should check first though. Change to @kaspth's suggestion and see if any test fail.